dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,156 @@
+      &iomuxc_gpio_ad_b0_13_lpuart1_rx>;
+    drive-strength = "r0-6";
+    slew-rate = "slow";
+    nxp,speed = "mhz-100";
+  };
+
+  Note that the mux, mode, input, daisy, and cfg values must be aligned for
+  correct configuration
+  This will select GPIO_AD_B0_12 as LPUART1 TX, and GPIO_AD_B0_13 as LPUART1 RX.
+  Both pins will be configured with a weak latch, drive strength of "r0-6",
+  slow slew rate, and 100 MHZ speed.
+  Note that the soc level iomuxc dts file can be examined to find the possible
+  pinmux options. Here are the affects of each property on the
+  IOMUXC SW_PAD_CTL register:
+  input-schmitt-enable: HYS=1
+  drive-open-drain: ODE=1
+  input-enable: SION=1 (in SW_MUX_CTL_PAD register)
+  bias-pull-down: PUE=1, PUS=<bias-pull-down-value>
+  bias-pull-up: PUE=1, PUS=<bias-pull-up-value>
+  bias-disable: PKE=0
+  slew-rate: SRE=<enum_idx>
+  drive-strength: DSE=<enum_idx>
+  nxp,speed: SPEED=<enum_idx>
+
+  If only required properties are supplied, the pin will have the following
+  configuration:
+  HYS=0,
+  ODE=0,
+  SION=0,
+  PUE=0,
+  PUS=0,
+  PKE=1,
+  SRE=<slew-rate>,
+  DSE=<drive-strength>,
+  SPEED=<nxp,speed>
+
+
+compatible: "nxp,mcux-rt-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - input-schmitt-enable
+            - drive-open-drain
+            - input-enable
+            - bias-disable
+            - bias-pull-down
+            - bias-pull-up
+
+child-binding:
+  description: MCUX RT pin controller pin group
+  child-binding:
+    description: |
+      MCUX RT pin controller pin configuration node.
+    properties:
+      pinmux:
+        required: true
+        type: phandles
+        description: |
+          Pin mux selections for this group. See the soc level iomuxc DTSI file
+          for a defined list of these options.
+      drive-strength:
+        required: true
+        type: string
+        enum:
+          - "disabled"
+          - "r0"
+          - "r0-2"
+          - "r0-3"
+          - "r0-4"
+          - "r0-5"
+          - "r0-6"
+          - "r0-7"
+        description: |
+          Pin output drive strength. Sets the DSE field in the IOMUXC peripheral.
+          the drive strength is expressed as a output impedance at a given voltage,
+          but maximum current values can be calculated from these impedances
+          for a specific load impedance.
+          000 DSE_0_output_driver_disabled_ — output driver disabled
+          001 DSE_1_R0_1 — 157 Ohm impedance @3.3V, 260 Ohm impedance @1.8V
+          010 DSE_2_R0_2 — 78 Ohm @3.3V, 130 Ohm @1.8V
+          011 DSE_3_R0_3 — 53 Ohm @3.3V, 88 Ohm @1.8V
+          100 DSE_4_R0_4 — 39 Ohm @3.3V, 65 Ohm @1.8V
+          101 DSE_5_R0_5 — 32 Ohm @3.3V, 52 Ohm @1.8V
+          110 DSE_6_R0_6 — 32 Ohm @3.3V, 43 Ohm @1.8V
+          111 DSE_7_R0_7 — 26 Ohm @3.3V, 37 Ohm @1.8V
```

Comment:
These values (`000` to `111`) should likely be removed since they are set using an enum.

Reply:
I personally think they are useful to have- the driver uses the enum IDX as the value to write to the register, and these values tell you exactly what will be written to the `DSE` register. That way a user looking at this file does not have to guess what effect `drive-strength="r0"` will have at the register level, or look into the driver code itself.